### PR TITLE
add filtering to Get-DbaDbAssembly

### PR DESCRIPTION
**Pull request by Stuart-Moore**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Adding encryption support to Backups
So need to add support for managing asymmetric keys
So need to add support for creating keys from assemblies
So need to check for a specific assembly in a specific db

So I made a change :)

No impact to anyone using the older parameter list

The others will follow along later to save a huge change

### Approach
Added the traditional dbatools filtering

### Commands to test
Example in code

### Learning
always add filtering to a command as at some point someone will want it to return a single object

### CODE REVIEW

**Comment by potatoqualitee** [comment]
encryption support to backups! 😍

**Comment by Stuart-Moore** [comment]
yep. It's a real rabbit hole once you start on it.
I've got it all working and the new dll compiled.
But to offer it as a real solution, dbatools needs to support all the options, which means a few tweaks to the certificate management commands and adding key support as well.

EKM support won't be in this release because that's tricky. But as I'm meant to be implementing it at work before Xmas using Azure Key Vault, there's a good chance it'll end up here at some point as well :)

File: functions/Get-DbaDbAssembly.ps1 [hunk 1]
```diff
@@ -59,16 +72,24 @@ function Get-DbaDbAssembly {
...
+            $databases = $server.Databases | Where-Object IsAccessible
+            if (Test-Bound 'Database') {
+                $databases = $databases | Where-Object Name -in $Database
+            }
+            foreach ($db in $databases) {
                 try {
-                    foreach ($assembly in $database.assemblies) {
+                    $assemblies = $db.assemblies
+                    if (Test-Bound 'Name') {
+                        $assemblies = $assemblies | Where-Object Name -in  $Name
```
Comment (potatoqualitee) [inline functions/Get-DbaDbAssembly.ps1]: this could potentially be if `test-bound name, $db.Assemblies[$name] else $db.assemblies`?

Comment (Stuart-Moore) [reply functions/Get-DbaDbAssembly.ps1]: Yep, I can change that. Was just copying from other functions.

Comment (Stuart-Moore) [reply functions/Get-DbaDbAssembly.ps1]: fixed in last commit

**Comment by wsmelton** [comment]
@Stuart-Moore when you merge please update the commit message to be formatted with `[Command] - [purpose]`....we are just trying to get a bit more consistent on commit messages to try and automate the change log a bit more.

**Comment by Stuart-Moore** [comment]
NP, sorry about that. Will remember for next time.

**Comment by wsmelton** [comment]
No worries....we just started doing it this week 😁

**Pull request merged**